### PR TITLE
Transformer toolkit updates

### PR DESCRIPTION
**Pull request by dirkgr**
This is spun out of some of the Tango stuff.

From the changelog:

- The activation layer in the transformer toolkit now can be queried for its output dimension.
- `TransformerEmbeddings` now takes, but ignores, a parameter for the attention mask. This is needed for compatibility with some other modules that get called the same way and use the mask.
- `TransformerPooler` can now be instantiated from a pretrained transformer module, just like the other modules in the transformer toolkit.

### CODE REVIEW

File: allennlp/modules/transformer/transformer_pooler.py [hunk 1]
```diff
@@ -1,11 +1,44 @@
...
         self,
         hidden_size: int,
         intermediate_size: int,
+        activation: Union[str, torch.nn.Module] = "relu",
     ):
-        super().__init__(hidden_size, intermediate_size, "relu", pool=True)
+        super().__init__(hidden_size, intermediate_size, activation, pool=True)
+
+    @classmethod
+    def _get_input_arguments(
```
Comment (AkshitaB) [inline allennlp/modules/transformer/transformer_pooler.py:25]: We don't require this method any longer. `from_config` takes care of what we need.

**Review by AkshitaB — CHANGES_REQUESTED** [review]

File: allennlp/modules/transformer/transformer_pooler.py (hunk 1 above)
Comment (dirkgr) [reply allennlp/modules/transformer/transformer_pooler.py]: Removed!

**Review by AkshitaB — APPROVED** [review]
